src/LangtailPrompts.spec.ts

```diff
@@ -42,100 +42,185 @@ describe(
+          true,
+        )
+        expect(completion.choices.length).toBeGreaterThan(0)
+        expect(completion.httpResponse.status).toBe(200)
       })
 
-      expect(completion.choices[0].message.content?.includes("Aragorn")).toBe(
-        true,
-      )
-      expect(completion.choices.length).toBeGreaterThan(0)
-      expect(completion.httpResponse.status).toBe(200)
-    })
+      it("should support streaming", async () => {
+        const proxyCompletion = await lt.invoke({
+          prompt,
+          environment: "staging",
+          variables: {
+            about: "napoleon",
+          },
+          stream: true,
+        })
+        let partCount = 0
+        for await (const part of proxyCompletion) {
+          partCount++
 
-    it("should support streaming", async () => {
-      const proxyCompletion = await lt.invoke({
-        prompt,
-        environment: "staging",
-        variables: {
-          about: "napoleon",
-        },
-        stream: true,
+          openAIStreamingResponseSchema.parse(part)
+        }
+
+        expect(partCount > 1).toBe(true)
       })
-      let partCount = 0
-      for await (const part of proxyCompletion) {
-        partCount++
 
-        openAIStreamingResponseSchema.parse(part)
-      }
+      it("should not record", async () => {
+        //full deployed prompt path "do-not-delete-api-key-used-on-ci-iSNqij/ci-tests-project/short-story-teller/staging"
+        const ltWithProject = new LangtailPrompts({
+          apiKey: process.env.LANGTAIL_API_KEY!,
+          workspace: "do-not-delete-api-key-used-on-ci-iSNqij",
+          project: "ci-tests-project",
+          fetch: async (url, init) => {
+            expect(init?.headers?.["x-langtail-do-not-record"]).toBe("true")
+            expect(init?.headers?.["x-langtail-metadata-custom-field"]).toBe(1)
+
+            return {
+              ok: true,
+              status: 200,
+              json: async () => ({
+                choices: [
+                  {
+                    message: {
+                      content: "This is a test",
+                    },
+                  },
+                ],
+              }),
+            } as any
+          },
+        })
 
-      expect(partCount > 1).toBe(true)
+        const res = await ltWithProject.invoke({
+          prompt,
+          environment: "staging",
+          variables: {
+            about: "napoleon",
+          },
+          doNotRecord: true,
+          metadata: {
+            "custom-field": 1,
+          },
+        })
+
+        expect(res.httpResponse.status).toBe(200)
+        // TODO when we have some API for logs we could assert on that too that the log record does not have any payload
+      })
     })
 
-    it("should not record", async () => {
-      //full deployed prompt path "do-not-delete-api-key-used-on-ci-iSNqij/ci-tests-project/short-story-teller/staging"
-      const ltWithProject = new LangtailPrompts({
+    describe.only("build", () => {
```

Comment:
Do not forget to remove `.only`.

Reply:
Yes. I will before I move this out of draft.